src/search_algo.hpp

```diff
@@ -529,6 +529,23 @@ search(LocalDataHolder<TGlobalHolder> & lH)
             if (seedBegin > (lH.redQrySeqs[i].size() - lH.options.seedLength))
                 break;
 
+            // Skip seed if it only contains the same letters in nucleotide and bisulfite mode
+            if constexpr (TGlobalHolder::c_origQryAlph == AlphabetEnum::DNA5)
+            {
+                bool all_equal = true;
+                for (size_t j = seedBegin + 1; j < (seedBegin + lH.options.seedLength); j++)
+                {
+                    if (lH.redQrySeqs[i][j] != lH.redQrySeqs[i][seedBegin])
+                    {
+                        all_equal = false;
+                        break;
+                    }
+                }
```

Comment:
This can be made a bit more readable with std::equal() or std::adjacent_find, but that's 💅🏻